src/sources/http_scrape/integration_tests.rs

```diff
@@ -0,0 +1,229 @@
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+    let log = events[0].as_log();
+    assert_eq!(log[log_schema().source_type_key()], NAME.into());
+}
+
+/// Metrics should be scraped and decoded successfully.
+#[tokio::test]
+async fn scraped_metrics_native_json() {
+    let events = run_compliance(HttpScrapeConfig::new(
+        "http://dufs:5000/metrics/native.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::NativeJson,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+
+    let metric = events[0].as_metric();
+    assert_eq!(
+        metric.tags().unwrap()[log_schema().source_type_key()],
+        NAME.to_string()
+    );
+}
+
+/// Traces should be scraped and decoded successfully.
+#[tokio::test]
+async fn scraped_trace_native_json() {
+    let events = run_compliance(HttpScrapeConfig::new(
+        "http://dufs:5000/traces/native.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::NativeJson,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+
+    let trace = events[0].as_trace();
+    assert_eq!(trace.as_map()[log_schema().source_type_key()], NAME.into());
+}
+
+/// Passing no authentication for the auth-gated endpoint should yield errors.
+#[tokio::test]
+async fn unauthorized_no_auth() {
+    run_error(HttpScrapeConfig::new(
+        "http://dufs-auth:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        None,
+    ))
+    .await;
+}
+
+/// Passing the incorrect credentials for the auth-gated endpoint should yield errors.
+#[tokio::test]
+async fn unauthorized_wrong_auth() {
+    run_error(HttpScrapeConfig::new(
+        "http://dufs-auth:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        Some(Auth::Basic {
+            user: "white_rabbit".to_string(),
+            password: "morpheus".to_string(),
+        }),
+    ))
+    .await;
+}
+
+/// Passing the correct credentials for the auth-gated endpoint should succeed.
+#[tokio::test]
+async fn authorized() {
+    run_compliance(HttpScrapeConfig::new(
+        "http://dufs-auth:5000/logs/json.json".to_string(),
+        INTERVAL_SECS,
+        None,
+        DeserializerConfig::Json,
+        default_framing_message_based(),
+        None,
+        None,
+        Some(Auth::Basic {
+            user: "user".to_string(),
+            password: "pass".to_string(),
+        }),
+    ))
+    .await;
+}
+
+/// Passing the CA file for TLS should yield errors.
```

Comment:
```suggestion
/// Passing an incorrect CA file for TLS should yield errors.
```

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/13793/commits/5f7c44231f801eb5ee0be2f2a3549efb6c908e8c

This also made me realize there was a bug in the test case